mailstream/mailstream.php

```diff
@@ -97,56 +98,58 @@ function mailstream_generate_id($a, $uri) {
 	return $message_id;
 }
 
-function mailstream_post_hook(&$a, &$item) {
+function mailstream_post_hook(App $a, array &$item) {
 	if (!PConfig::get($item['uid'], 'mailstream', 'enabled')) {
 		return;
 	}
-	if (!$item['uid']) {
-		return;
-	}
-	if (!$item['contact-id']) {
+
+	if (!isset($item['uid']) || !isset($item['contact-id']) || !isset($item['uri'])) {
 		return;
 	}
-	if (!$item['uri']) {
+
+	if (PConfig::get($item['uid'], 'mailstream', 'nolikes') && $item['verb'] == ACTIVITY_LIKE) {
 		return;
 	}
-	if (PConfig::get($item['uid'], 'mailstream', 'nolikes')) {
-		if ($item['verb'] == ACTIVITY_LIKE) {
-			return;
-		}
-	}
 
 	$message_id = mailstream_generate_id($a, $item['uri']);
 	q("INSERT INTO `mailstream_item` (`uid`, `contact-id`, `uri`, `message-id`) " .
 		"VALUES (%d, '%s', '%s', '%s')", intval($item['uid']),
 		intval($item['contact-id']), dbesc($item['uri']), dbesc($message_id));
 	$r = q('SELECT * FROM `mailstream_item` WHERE `uid` = %d AND `contact-id` = %d AND `uri` = "%s"', intval($item['uid']), intval($item['contact-id']), dbesc($item['uri']));
-	if (count($r) != 1) {
+
+	if (!DBM::is_result($r)) {
```

Comment:
This should be rewritten with a `dba::exists()` call.

Reply:
Should I add a TODO here for next PR?

Reply:
Scratch that, it should be a `dba::selectFirst` instead. You can leave it out for later.

Reply:
Sure.